compiler/src/main/kotlin/ast/symbol.kt

```diff
@@ -79,4 +105,153 @@ class SymbolTable {
     fun openScope(index: Int) {
         scopeStack.push(scopeStack.peek().tables[index])
     }
+}
+
+/**
+ * Assists in building and filling a symbol table
```

Comment:
Does not describe how it works at all. Maybe something like the following, but more elaborate:

It uses a stack, where tables are pushed and popped from, searches all tables on stack for getting, etc.

Reply:
The documentation could maybe be expanded, but I would like keep implementation details out of the javadoc since users of the class are not supposed to rely on the internal implementation.
I'm open to adding a comment inside the class that describe how it's supposed to work and how it's implemented.